pkg/cloud/services/autoscaling/autoscalinggroup.go

```diff
@@ -465,3 +460,38 @@ func mapToTags(input map[string]string, resourceID *string) []*autoscaling.Tag {
 	}
 	return tags
 }
+
+// SubnetIDs return subnets IDs of a AWSMachinePool based on given subnetIDs and filters.
+func (s *Service) SubnetIDs(scope *scope.MachinePoolScope) ([]string, error) {
+	subnetIDs := make([]string, len(scope.AWSMachinePool.Spec.Subnets))
+	for i, v := range scope.AWSMachinePool.Spec.Subnets {
+		subnetIDs[i] = aws.StringValue(v.ID)
```

Comment:
This will introduce the problem mentioned in this issue https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2087.
Hence we should take care of subnet lookup here before assigning the subnetID value directly.

Reply:
But can be taken care as part of the above mentioned issue as well.